### PR TITLE
Remove Certificate Creation From RNW App Template

### LINKED ISSUE #8202: Do not create temporary certs in default template
We've been doing work to remove temporary signing certs from our repo. Apart from security goodness, these checked in certs expire and leave the world unable to deploy. FURN just recently hit this.

We should avoid temporary certs in the default case.

### PR DESCRIPTION
**Pull request by chiaramooney**
**_Why_**
Certificates should not be publicly published on this repository. We do not want developers to use a certificate with themselves not as the publisher (current certificate generation code contains case where default certificate in repo is used for newly created app). Self-signed certificates expire quickly leaving users with a sudden break in building approx a year after project creation.

Goal here is to remove ourselves from certificate handling/creation entirely for RNW users. Instead, responsibility will be put on users to generate a certificate on their end when they are ready to publish their work. 

**_What_**
Removed remaining certificate instances in repository. Removed certificate creation from app template. 

Note change will not affect Debug builds and deploys as long as developer does not set MSBuild prop AppxSigningEnabled to true. For Release, builds will work with out a certificate, but if the developer wishes to deploy they will need to either use the --deploy-from-layout option or generate a certificate for themselves in Visual Studio. 

**_Future Work_**
Once 0.66 website docs are cut (since this change will not be seen until 0.67), I will make an update to the docs to note the change in the template and alert developers that they must now create a certificate if they wish to deploy in Release mode without deploying from layout. 

Resolves #8202

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8672)

### CODE REVIEW

**Review by dannyvv — APPROVED** [review]

**Comment by dannyvv** [comment]
Does UWP give a good actionable error message when trying to publish without certs?

**Comment by chiaramooney** [comment]
> Does UWP give a good actionable error message when trying to publish without certs?

You will definitely get an error that you are missing a cert when you try to build an AppxPackage via the cli. But I don't believe the error message lists or links steps to how to generate a cert to resolve the issue. This is where having clear/discoverable documentation on the website will be important. If you try to create an AppxPackage via VS, it will prompt you to select a cert / walk you through the options of creating one.

Another thing to note, is that the self-signed cert that was originally generated by the template prior to this change cannot be used to publish an app to the Microsoft Store. It's only purpose is for local testing, so regardless developers looking to publish to the store will need to make the investment to research about certificates.

File: .ado/jobs/cli-init.yml [hunk 1]
```diff
@@ -54,14 +54,14 @@ jobs:
             platform: x64
             projectType: app
             additionalInitArguments:
-            additionalRunArguments:
+            additionalRunArguments: --deploy-from-layout
```
Comment (jonthysell) [inline .ado/jobs/cli-init.yml:57]: So are we no longer testing CLI release with certs?

**Review by jonthysell — APPROVED** [review]

File: .ado/jobs/cli-init.yml (hunk 1 above)
Comment (chiaramooney) [reply .ado/jobs/cli-init.yml:57]: We could if we wanted to in CI! I can do the same process as I did for other release builds and pull in a cert, but we won't have access to this in PR pipelines.

**Pull request merged**